backend/src/main/java/backend/service/impl/UtilsServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package backend.service.impl;
+
+import backend.entity.User;
+import backend.exception.not_found.WikipediaDefinitionNotFound;
+import backend.exception.unauthorized.UnauthorizedException;
+import backend.exception.unauthorized.UserNotFoundException;
+import backend.repository.UserRepository;
+import backend.security.TokenAuthentication;
+import backend.utils.WikipediaUtils;
+import backend.service.UtilsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class UtilsServiceImpl implements UtilsService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private TokenAuthentication tokenAuthentication;
+
+    @Override
+    public String getWikipediaDefinition(String wantedText, String authorizationToken) throws WikipediaDefinitionNotFound, UserNotFoundException {
+        String currentUserName = tokenAuthentication.getUsername(authorizationToken);
+        Optional<User> currentUser = userRepository.findByUsernameIgnoreCase(currentUserName);
+        if (currentUser.isPresent()) {
+            String result = WikipediaUtils.getWikipediaDefinition(wantedText, currentUser.get().getLanguage());
```

Comment:
To nie zadziała.
Z wikipedii korzystają redaktotrzy, którzy nie mają przypisanego języka.
Z tego co kojarzę to nie musimy obsługiwać innych języków wikipedii

Reply:
ok w takim razie przerobiłem, defaultowy język to pl, ale można w parametrze przekazać ang w razie potrzeby